core/src/main/java/com/microsoft/applicationinsights/internal/util/CollectionTypeJsonSerializer.java

```diff
@@ -0,0 +1,38 @@
+package com.microsoft.applicationinsights.internal.util;
+
+import com.fasterxml.jackson.databind.BeanDescription;
+import com.fasterxml.jackson.databind.JavaType;
+import com.fasterxml.jackson.databind.JsonSerializer;
+import com.fasterxml.jackson.databind.SerializationConfig;
+import com.fasterxml.jackson.databind.jsontype.TypeSerializer;
+import com.fasterxml.jackson.databind.module.SimpleSerializers;
+import com.fasterxml.jackson.databind.type.CollectionType;
+import java.util.List;
+
+public class CollectionTypeJsonSerializer extends SimpleSerializers {
+
+    @Override
+    public JsonSerializer<?> findCollectionSerializer(SerializationConfig config,
+                                                      CollectionType type,
+                                                      BeanDescription beanDesc,
+                                                      TypeSerializer elementTypeSerializer,
+                                                      JsonSerializer<Object> elementValueSerializer) {
+
+        if (isJavaLangObjectListType(type)) {
+            return new NdJsonSerializer();
+        }
+        return super.findCollectionSerializer(config, type, beanDesc, elementTypeSerializer, elementValueSerializer);
+    }
+
+
+    private boolean isJavaLangObjectListType(CollectionType type) {
+        JavaType contentType = type.getContentType();
+        if (List.class.isAssignableFrom(type.getRawClass())) {
+            // this means it's a list
+            if(contentType.isJavaLangObject()) {
```

Comment:
I thought this would be checking contentType is `TelemetryItem`?

Reply:
The list contains [simple type, class java.lang.Object], is checking contentType as TelemetryItem possible in this case?